File: testdata/testkit/kotlin/src/main/kotlin/domain/Entity.kt
package domain

data class Entity(val id: String, val name: String, val data: Map<String, String> = emptyMap())

interface Repository {
    suspend fun findById(id: String): Entity?
    suspend fun save(entity: Entity)
}

sealed class Result<out T> {
    data class Success<T>(val value: T) : Result<T>()
    data class Failure(val error: String) : Result<Nothing>()
}

File: testdata/testkit/typescript/src/domain/entity.ts
export interface Entity {
  id: string;
  name: string;
  data: Record<string, string>;
}

export interface Repository {
  findById(id: string): Promise<Entity | null>;
  save(entity: Entity): Promise<void>;
}

export class Service {
  constructor(private readonly repo: Repository) {}

  async getEntity(id: string): Promise<Entity | null> {
    return this.repo.findById(id);
  }
}
